[FreeBSD] Compilation of libfuse fails on FreeBSD due to an unrecognized type/codes

FAILED: example/notify_inval_inode.p/notify_inval_inode.c.o
cc -Iexample/notify_inval_inode.p -Iexample -I../example -Iinclude -I../include -Ilib -I../lib -I. -I.. -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -O2 -g -D_REENTRANT -DHAVE_LIBFUSE_PRIVATE_CONFIG_H -Wno-sign-compare -D_FILE_OFFSET_BITS=64 -Wstrict-prototypes -Wmissing-declarations -Wwrite-strings -fno-strict-aliasing -pthread -MD -MQ example/notify_inval_inode.p/notify_inval_inode.c.o -MF example/notify_inval_inode.p/notify_inval_inode.c.o.d -o example/notify_inval_inode.p/notify_inval_inode.c.o -c ../example/notify_inval_inode.c
**../example/notify_inval_inode.c:289:44: error: use of undeclared identifier 'EBADFD'**
                 ret != -ENOENT && ret != -EBADFD && ret != -ENODEV) {

.**/example/notify_store_retrieve.c:355:43: error: use of undeclared identifier 'EBADFD'**
                ret != -ENOENT && ret != -EBADFD && ret != -ENODEV) {


[20/62] Compiling C object example/hello_ll_uds.p/hello_ll_uds.c.o
../example/hello_ll_uds.c:287:11: **warning: implicit declaration of function 'splice' is invalid in C99** [-Wimplicit-function-declaration]
                int i = splice(fdin, offin, fdout, offout, len - count, flags);

2/13] Compiling C object test/test_write_cache.p/test_write_cache.c.o
FAILED: test/test_write_cache.p/test_write_cache.c.o
cc -Itest/test_write_cache.p -Itest -I../test -Iinclude -I../include -Ilib -I../lib -I. -I.. -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -O2 -g -D_REENTRANT -DHAVE_LIBFUSE_PRIVATE_CONFIG_H -Wno-sign-compare -D_FILE_OFFSET_BITS=64 -Wstrict-prototypes -Wmissing-declarations -Wwrite-strings -fno-strict-aliasing -pthread -MD -MQ test/test_write_cache.p/test_write_cache.c.o -MF test/test_write_cache.p/test_write_cache.c.o.d -o test/test_write_cache.p/test_write_cache.c.o -c ../test/test_write_cache.c
.**./test/test_write_cache.c:217:5: error: unknown type name 'loff_t'; did you mean 'off_t'?**
    loff_t off = 0;
    ^~~~~~

--- a/example/meson.build
+++ b/example/meson.build
@@ -1,10 +1,10 @@
 examples = [ 'passthrough', 'passthrough_fh',
-             'hello', 'hello_ll', 'hello_ll_uds',
+             'hello', 'hello_ll',
              'printcap', 'ioctl_client', 'poll_client',
              'ioctl', 'cuse', 'cuse_client' ]
 
 if not platform.endswith('bsd') and platform != 'dragonfly'
-    examples += 'passthrough_ll'
+    examples += [ 'passthrough_ll', 'hello_ll_uds' ]
 
     # According to Conrad Meyer <cem@freebsd.org>, FreeBSD doesn't
     # support mounting files, This is enforced in vfs_domount_first()

--- a/example/notify_inval_inode.c
+++ b/example/notify_inval_inode.c
@@ -287,14 +287,14 @@ static void* update_fs_loop(void *data) {
         if (!options.no_notify && lookup_cnt) {
             /* Only send notification if the kernel is aware of the inode */
 
-            /* Some errors (ENOENT, EBADFD, ENODEV) have to be accepted as the
+            /* Some errors (ENOENT, EBADF, ENODEV) have to be accepted as they
              * might come up during umount, when kernel side already releases
              * all inodes, but does not send FUSE_DESTROY yet.
              */
             int ret =
                 fuse_lowlevel_notify_inval_inode(se, FILE_INO, 0, 0);
             if ((ret != 0 && !is_stop) &&
-                 ret != -ENOENT && ret != -EBADFD && ret != -ENODEV) {
+                 ret != -ENOENT && ret != -EBADF && ret != -ENODEV) {
                 fprintf(stderr,
                         "ERROR: fuse_lowlevel_notify_store() failed with %s (%d)\n",
                         strerror(-ret), -ret);

--- a/example/notify_store_retrieve.c
+++ b/example/notify_store_retrieve.c
@@ -353,14 +353,14 @@ static void* update_fs_loop(void *data) {
             bufv.buf[0].flags = 0;
 
             /*
-             * Some errors (ENOENT, EBADFD, ENODEV) have to be accepted as they
+             * Some errors (ENOENT, EBADF, ENODEV) have to be accepted as they
              * might come up during umount, when kernel side already releases
              * all inodes, but does not send FUSE_DESTROY yet.
              */
 
             ret = fuse_lowlevel_notify_store(se, FILE_INO, 0, &bufv, 0);
             if ((ret != 0 && !is_umount) &&
-                ret != -ENOENT && ret != -EBADFD && ret != -ENODEV) {
+                ret != -ENOENT && ret != -EBADF && ret != -ENODEV) {
                 fprintf(stderr,
                         "ERROR: fuse_lowlevel_notify_store() failed with %s (%d)\n",
                         strerror(-ret), -ret);
@@ -371,7 +371,7 @@ static void* update_fs_loop(void *data) {
                kernel to send us back the stored data */
             ret = fuse_lowlevel_notify_retrieve(se, FILE_INO, MAX_STR_LEN,
                                                 0, (void*) strdup(file_contents));
-            assert((ret == 0 || is_umount) || ret == -ENOENT || ret == -EBADFD ||
+            assert((ret == 0 || is_umount) || ret == -ENOENT || ret == -EBADF ||
                    ret != -ENODEV);
             if(retrieve_status == 0)
                 retrieve_status = 1;

--- a/test/test_write_cache.c
+++ b/test/test_write_cache.c
@@ -9,8 +9,6 @@
 
 #define FUSE_USE_VERSION 30
 
-#define _GNU_SOURCE /* for loff_t */
-
 #include <fuse_config.h>
 #include <fuse_lowlevel.h>
 #include <stdio.h>
@@ -214,7 +212,7 @@ static void test_fs(char *mountpoint) {
     const size_t dsize = options.data_size * WRITE_SYSCALLS;
     int fd, rofd;
     pthread_t rofd_thread;
-    loff_t off = 0;
+    off_t off = 0;
 
     buf = malloc(dsize);
     assert(buf != NULL);

